zhaquirks/tuya/__init__.py

```diff
@@ -1627,30 +1630,40 @@ def handle_set_time_request(self, payload: t.uint16_t) -> foundation.Status:
         """Handle Time set request."""
         return foundation.Status.SUCCESS
 
+    @property
+    def _dp_to_attributes(self) -> dict[int, list[DPToAttributeMapping]]:
+        """Convert from legacy single attribute dp mappings."""
+        return {
+            dp: attr if isinstance(attr, list) else [attr]
+            for dp, attr in self.dp_to_attribute.items()
+        }
```

Comment:
Hmm, this will be "regenerated" every time it's accessed and that'll be quite often.
We might be able to make this a `cached_property`, but I'd prefer if we can convert this in when initializing (in `__init__` of this cluster?).
I'm not sure if we have quirks that change `dp_to_attribute` at runtime though. Probably unlikely, but we'll also have to check that we don't break the Tuya quirk builder.

I'll get back to this.

Reply:
moved to `__init__`. The Tuya builder sets `dp_to_attribute` when defining the class, before initialization, so it should be ok.